includes/REST_API/Stories_Controller.php

```diff
@@ -270,140 +289,160 @@ public function get_items( $request ) {
 			}
 
-			if ( ! empty( $request[ $tax_exclude ] ) ) {
-				$query_args['tax_query'][] = [
-					'taxonomy'         => $taxonomy->name,
-					'field'            => 'term_id',
-					'terms'            => $request[ $tax_exclude ],
-					'include_children' => false,
-					'operator'         => 'NOT IN',
-				];
+			if ( $tax_exclude ) {
+				$terms            = [];
+				$include_children = false;
+
+				if ( rest_is_array( $tax_exclude ) ) {
+					$terms = $tax_exclude;
+				} elseif ( rest_is_object( $tax_exclude ) ) {
+					$terms            = empty( $tax_exclude['terms'] ) ? [] : $tax_exclude['terms'];
+					$include_children = ! empty( $tax_exclude['include_children'] );
+				}
+
+				if ( $terms ) {
+					$args['tax_query'][] = [
+						'taxonomy'         => $taxonomy->name,
+						'field'            => 'term_id',
+						'terms'            => $terms,
+						'include_children' => $include_children,
+						'operator'         => 'NOT IN',
+					];
+				}
 			}
 		}
 
+		return $args;
+	}
+
+	/**
+	 * Add response headers, with post counts.
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 *
+	 * @since 1.12.0
+	 *
+	 * @param WP_REST_Response $response Response object.
+	 * @param WP_REST_Request  $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error
+	 */
+	protected function add_response_headers( WP_REST_Response $response, WP_REST_Request $request ) {
 		// Add counts for other statuses.
 		$statuses = [
-			'all'     => [ 'publish' ],
 			'publish' => 'publish',
 		];
 
 		if ( $this->get_post_type_cap( $this->post_type, 'edit_posts' ) ) {
-			$statuses['all'][]  = 'draft';
-			$statuses['all'][]  = 'future';
 			$statuses['draft']  = 'draft';
 			$statuses['future'] = 'future';
 		}
 
-		if ( $this->get_post_type_cap( $this->post_type, 'read_private_posts' ) ) {
-			$statuses['all'][]   = 'private';
+		if ( $this->get_post_type_cap( $this->post_type, 'publish_posts' ) ) {
 			$statuses['private'] = 'private';
 		}
 
-		$statuses_count = [];
+		$edit_others_posts  = $this->get_post_type_cap( $this->post_type, 'edit_others_posts' );
+		$edit_private_posts = $this->get_post_type_cap( $this->post_type, 'edit_private_posts' );
+
+		$statuses_count = [ 'all' => 0 ];
+		$total_posts    = 0;
+
+		$query_args = $this->prepare_items_query( $this->args, $request );
 
 		// Strip down query for speed.
 		$query_args['fields']                 = 'ids';
 		$query_args['posts_per_page']         = 1;
+		$query_args['paged']                  = 1;
 		$query_args['update_post_meta_cache'] = false;
 		$query_args['update_post_term_cache'] = false;
 
 		foreach ( $statuses as $key => $status ) {
 			$posts_query               = new WP_Query();
 			$query_args['post_status'] = $status;
+			if ( in_array( $status, [ 'draft', 'future' ], true ) && ! $edit_others_posts ) {
+				$query_args['author'] = get_current_user_id();
+			}
+			if ( 'private' === $status && ! $edit_private_posts ) {
+				$query_args['author'] = get_current_user_id();
+			}
```

Comment:
Can we remove the author filtering again as per our discussion?

Users should see all stories by default, not only their own ones.

We can then create a follow-up ticket to further improve this.

I'll try to take a look at the WP core code you shared too so I get a better picture of this.

Reply:
These filters are only for the counts and not displaying the story. The filter of quests was removed when I remove this commit - https://github.com/google/web-stories-wp/commit/7ff521af114ee21ea033b4fa0b595bb3bc5fcb6b